### PR TITLE
Add Google Analytics tracking code

### LINKED ISSUE #329: Add analytics around page views and which modes are being used
- Set up a new property for the damage calc in Google Analytics
- add something like following snippet:

```html
<!-- Google Analytics -->
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','//www.google-analytics.com/analytics.js','ga');
ga('create', 'UA-XXXXXX-X', 'pokemonshowdown.com/damagecalc');
ga('send', 'pageview');
</script>
<!-- End Google Analytics -->
<script>
	window.onerror = function(err, uri, line) {
		var msg = uri + ':' + line + ': ' + err;
		if ('ga' in window) {
			ga('send', 'event', 'Client error', msg);
		} else {
			alert('ERROR: ' + msg);
		}
		return false;
	};
</script>
```

Though we also want to track:
  -  `generation`
  - `mode` (one vs one, one vs all, all vs one)
  - `format` (standard, vgc, random, bss, etc)

### PR DESCRIPTION
**Pull request by scheibo**
Fixes #329.

### CODE REVIEW

File: src/honkalculate.template.html [hunk 1]
```diff
@@ -9,6 +9,24 @@
...
     <link type="text/css" rel="stylesheet" href="./js/vendor/select2/select2.css" />
     <link type="text/css" rel="stylesheet" href="./css/main.css?" />
+    <!-- Google Analytics -->
+    <script>
+        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+        })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
+
+        ga('create', 'UA-XXXXXX-Y', 'calc.pokemonshowdown.com');
```
Comment (scheibo) [inline src/honkalculate.template.html]: @Zarel - please set up the property for this.

**Comment by scheibo** [comment]
We should be able to get `mode` from the `pageview` because its a query param, but I've added an event to track `gen`. Though we won't be getting both `gen` and `mode` linked, so maybe we should change that?

File: src/honkalculate.template.html (hunk 1 above)
Comment (Zarel) [reply src/honkalculate.template.html]: We got clogged up on whether or not this should be a subdomain or subdirectory. I guess separate property is fine, though.

**Comment by Zarel** [comment]
I set up the property. Analytics tells you to use this code:

```html
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-26211653-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-26211653-3');
</script>
```

File: src/honkalculate.template.html (hunk 1 above)
Comment (scheibo) [reply src/honkalculate.template.html]: You had [agreed to calc.pokemonshowdown.com](https://discordapp.com/channels/@me/551083939561865238/607642014280056839), and the rename issue (#332) assigned to you covers moving to a subdomain.

Comment (Zarel) [reply src/honkalculate.template.html]: Yes, that's why separate property is ifne.

**Comment by scheibo** [comment]
> Analytics tells you to use this code:

Fine, whatever, we'll use the new hotness.

I removed the `gen` stuff because I want to switch to `calc.pokemonshowdown.com/gen1` etc as discussed on Discord and i'll add tracking for that then (or maybe we'd then get it simply through the pageviews metric)

**Comment by AustinXII** [comment]
:+1 now we wait

**Comment by Zarel** [comment]
Are we not listening for client errors, then?

**Comment by scheibo** [comment]
It's supposed to be automatic with google tag manager - you just do setup in the UI AFAICT.

**Comment by Zarel** [comment]
Can you do the setup? (You should have access now.)

**Comment by Zarel** [comment]
> It's supposed to be automatic with google tag manager - you just do setup in the UI AFAICT.

https://developers.google.com/analytics/devguides/collection/gtagjs/exceptions

It does not appear to be automatic. Am I missing something?

**Comment by scheibo** [comment]
I confused Google Tag Manager a www.googletagmanager.com/gtag/js. I wonder how?

**Pull request merged**